language/compiler/ir_to_bytecode/syntax/src/syntax.rs

```diff
@@ -1 +1,1798 @@
+
+    // Continue parsing binary expressions as long as they have they
+    // specified minimum precedence.
+    while next_tok_prec >= min_prec {
+        let op_token = (tokens.token.1).0;
+        tokens.advance()?;
+
+        let mut rhs = parse_unary_exp_(tokens)?;
+
+        // If the next token is another binary operator with a higher
+        // precedence, then recursively parse that expression as the RHS.
+        let this_prec = next_tok_prec;
+        next_tok_prec = get_precedence(&(tokens.token.1).0);
+        if this_prec < next_tok_prec {
+            rhs = parse_rhs_of_binary_exp(tokens, rhs, this_prec + 1)?;
+            next_tok_prec = get_precedence(&(tokens.token.1).0);
+        }
+
+        let op = match op_token {
+            Tok::EqualEqual => BinOp::Eq,
+            Tok::ExclaimEqual => BinOp::Neq,
+            Tok::Less => BinOp::Lt,
+            Tok::Greater => BinOp::Gt,
+            Tok::LessEqual => BinOp::Le,
+            Tok::GreaterEqual => BinOp::Ge,
+            Tok::PipePipe => BinOp::Or,
+            Tok::AmpAmp => BinOp::And,
+            Tok::Caret => BinOp::Xor,
+            Tok::Pipe => BinOp::BitOr,
+            Tok::Amp => BinOp::BitAnd,
+            Tok::Plus => BinOp::Add,
+            Tok::Minus => BinOp::Sub,
+            Tok::Star => BinOp::Mul,
+            Tok::Slash => BinOp::Div,
+            Tok::Percent => BinOp::Mod,
+            _ => panic!("Unexpected token that is not a binary operator"),
+        };
+        let start_loc = result.span.start();
+        let end_loc = tokens.token.0;
+        let e = Exp::BinopExp(Box::new(result), op, Box::new(rhs));
+        result = Spanned{span: Span::new(start_loc, ByteIndex(end_loc as u32)),
+                         value: e };
+    }
+
+    Ok(result)
+}
+
+// QualifiedFunctionName : FunctionCall = {
+//     <f: Builtin> => FunctionCall::Builtin(f),
+//     <module_dot_name: DotName> <type_actuals: TypeActuals> =>? { ... }
+// }
+
+fn parse_qualified_function_name_<'input, 'builder>(tokens: &mut __Matcher<'input, 'builder>)
+    -> Result<FunctionCall_, ParseError<usize, Token<'input>, failure::Error>>
+{
+    let start_loc = tokens.token.0;
+    let call = match (tokens.token.1).0 {
+        Tok::CreateAccount |
+        Tok::Exists |
+        Tok::BorrowGlobal |
+        Tok::BorrowGlobalMut |
+        Tok::GetTxnGasUnitPrice |
+        Tok::GetTxnMaxGasUnits |
+        Tok::GetTxnPublicKey |
+        Tok::GetTxnSender |
+        Tok::GetTxnSequenceNumber |
+        Tok::MoveFrom |
+        Tok::MoveToSender |
+        Tok::GetGasRemaining |
+        Tok::Freeze => {
+            let f = parse_builtin(tokens)?;
+            FunctionCall::Builtin(f)
+        },
+        Tok::DotNameValue => {
+            let module_dot_name = parse_dot_name(tokens)?;
+            let type_actuals = parse_type_actuals(tokens)?;
+            let v: Vec<&str> = module_dot_name.split(".").collect();
+            assert!(v.len() == 2, 42);
```

Comment:
42? This looks like a Move assert :)

Reply:
I have no idea. I just copied that from the old lalrpop file. It certainly doesn't seem like a good idea to have these hardcoded constants, but we should make a separate change to replace them.